spec/support/features/javascript_driver_helper.rb

```diff
@@ -2,4 +2,24 @@ module JavascriptDriverHelper
   def javascript_enabled?
     %i[headless_chrome headless_chrome_mobile].include?(Capybara.current_driver)
   end
+
+  def with_awaited_fetch
+    setup_js = <<~JS
+      window._fetch = window.fetch;
+      window.fetch = async (...args) => {
+        window.isFetching = true;
+        const result = await window._fetch.call(window, ...args);
+        window.isFetching = false;
+        return result;
+      };
+    JS
+    teardown_js = 'window.fetch = window._fetch; delete window._fetch;'
+
+    page.execute_script(setup_js)
+    yield
+    Timeout.timeout(Capybara.default_max_wait_time) do
+      loop if page.evaluate_script('window.isFetching')
+    end
+    page.execute_script(teardown_js)
+  end
```

Comment:
I love this helper! I wonder if would help with the [mock device profiling code I just added](https://github.com/18F/identity-idp/blob/688a0d104f23296459a3316bb5282e691d24c360/spec/features/idv/doc_auth/ssn_step_spec.rb#L31)

if this was pure ruby code, I'd suggest an `ensure` for the best cleanup, since that way if the timeout throws an exception it's cleaned up. But I guess that each new page load will get a fresh `window.fetch`... so maybe cleaning up is not a priority

```suggestion
  ensure
    page.execute_script(teardown_js)
  end
```

Reply:
Yep, I think that's a good call 👍